app-ng/src/entry-point.lisp

```diff
@@ -4,31 +4,102 @@
 
 (in-package #:qvm-app-ng)
 
-(defparameter *option-spec* '())
+(defparameter *option-spec*
+  `((("help" #\h)
+     :type boolean
+     :optional t
+     :documentation "Display help")
+    (("host")
+     :type string
+     :initial-value ,+default-server-address+
+     :documentation "Host on which to start the QVM server")
+    (("log-level")
+     :type string
+     :initial-value "debug"
+     :documentation ,(format nil "Maximum logging level. Must be one of ~{~(~A~)~^, ~}"
+                             +available-log-levels+))
+    (("port" #\p)
+     :type integer
+     :initial-value ,+default-server-port+
+     :documentation "Port to start the QVM server on")
+    (("rpc-request")
+     :type string
+     :initial-value nil
+     :documentation "An initial RPC request to execute on application startup. The value should be a JSON object that corresponds to the request body you would send to invoke the RPC method via the HTTP server. For example, to invoke the \"version\" method, pass '{\"type\":\"version\"}'. The result will printed on *STANDARD-OUTPUT*.")
+    (("server" #\S)
+     :type boolean
+     :optional t
+     :initial-value nil
+     :documentation "Start a QVM server")
+    (("verbose")
+     :type integer
+     :initial-value 1
+     :documentation "Desired verbosity level. The following levels are understood.
+
+> 1: verbose output
+= 1: normal logging and output (default)
+< 1: no logging and only essential output")
+    (("version" #\v)
+     :type boolean
+     :optional t
+     :documentation "Display the versions of the app and underlying QVM")))
+
+(defun process-options (&key
+                          help
+                          host
+                          log-level
+                          port
+                          rpc-request
+                          server
+                          verbose
+                          version)
+
+  (when help
+    (show-help)
+    (return-from process-options))
+
+  (when version
+    (show-version)
+    (return-from process-options))
+
+  (when (> verbose 0)
+    (setf *logger* (make-logger "qvm-ng" (parse-log-level log-level)))
+    (show-welcome))
+
+  (when rpc-request
+    (run-initial-rpc-request rpc-request))
+
+  (when server
+    (start-server-mode :host host :port port)
+    #|not reached|#))
 
 (defun show-help ()
-  (format t "Usage:~%")
-  (format t "    qvm-ng [<options>...]~%~%")
-  (format t "Options:~%")
-  (command-line-arguments:show-option-help *option-spec* :sort-names t))
+  (format *error-output* "Usage:~%")
+  (format *error-output* "    qvm-ng [<options>...]~%~%")
+  (format *error-output* "Options:~%")
+  (command-line-arguments:show-option-help *option-spec* :stream *error-output* :sort-names t))
 
 (defun show-version ()
-  (format t "~A [~A]~%" +QVM-VERSION+ +GIT-HASH+))
+  (format *error-output* "~A [~A]~%" +QVM-VERSION+ +GIT-HASH+))
```

Comment:
is this usual practice?

Reply:
Probably not, no. I think I originally switched only `show-welcome` in order to not have the welcome message intermingle with other output, then switched `show-help` and `show-version` without giving it much though. But since both of those exit immediately after, and since help/version is the output the user requested, I makes sense to put that on stdout.

Reply:
Latest commit switches back to `*standard-output*` for `show-version` and `show-help`.